### PR TITLE
feat(emptyscreen): Empty Screen for Showing Errors or Blank Pages

### PR DESCRIPTION
**Pull request by ChiragJain-Haptik**
### JIRA Ticket

Jira:  [BB-5196](https://hello-haptik.atlassian.net/browse/BB-5196)

--

### Changelog

- An Empty Screen for showing errors and blank pages.
- One can add Title, Subtitle and Icon.
- It will take the complete width of parent div.

--

### Design
- [Design Link](https://www.figma.com/file/yTavkdrfIsMctbOVtHy02W/Master-Prototype?node-id=523%3A2549) - Build(CS) 6.12

### Screenshots:
image

### CODE REVIEW

File: src/components/UI/EmptyScreen/StyledEmptyScreen.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export const MessageContainer = styled(Box)({
+  display: "flex",
+  flexDirection: "column",
+  alignItems: "center",
+});
+
+export const IconBox = styled(Icon)({
+  minWidth: "100px",
+  height: "auto",
+  display: "table",
```
Comment (sidMulay) [inline src/components/UI/EmptyScreen/StyledEmptyScreen.js:30]: Why have we used `table` here?

File: src/components/UI/EmptyScreen/EmptyScreen.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import Box from "@src/components/UI/Box";
+import Text from "@src/components/UI/Text";
+
+import { ScreenWrapper, MessageContainer, IconBox } from "./StyledEmptyScreen";
+
+const EmptyScreen = ({ title, subtitle, icon }) => {
```
[the remark this answers is not in the document]
Comment (ChiragJain-Haptik) [reply src/components/UI/EmptyScreen/EmptyScreen.js:9]: Yes, right
Can you suggest a name?

File: src/components/UI/EmptyScreen/EmptyScreen.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  return (
+    <ScreenWrapper>
+      <MessageContainer>
+        <IconBox src={icon} />
```
[the remark this answers is not in the document]
Comment (ChiragJain-Haptik) [reply src/components/UI/EmptyScreen/EmptyScreen.js:13]: I have kept icon as compulsory. This is discussed with Shahnawaz, so he said to keep that compulsory to show the message with more details.

File: src/components/UI/EmptyScreen/EmptyScreen.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        <Box justifyContent="center">
+          <Text
+            data-testid="title"
+            variant="display-2.semi-bold"
+            color="black.haptik"
+            textAlign="center"
+            lineHeight="29px"
+          >
+            {title}
```
[the remark this answers is not in the document]
Comment (ChiragJain-Haptik) [reply src/components/UI/EmptyScreen/EmptyScreen.js:22]: Yes, title is also compulsory

File: src/components/UI/EmptyScreen/StyledEmptyScreen.js (hunk 1 above)
Comment (ChiragJain-Haptik) [reply src/components/UI/EmptyScreen/StyledEmptyScreen.js:30]: To make the icon position always center

File: src/components/UI/EmptyScreen/EmptyScreen.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            <Text
+              data-testid="subtitle"
+              variant="body.lg"
+              color="black.haptik"
+              textAlign="center"
+              lineHeight="19px"
+            >
+              {subtitle}
+            </Text>
+          </Box>
```
[the remark this answers is not in the document]
Comment (ChiragJain-Haptik) [reply src/components/UI/EmptyScreen/EmptyScreen.js:44]: Done

File: src/components/UI/EmptyScreen/StyledEmptyScreen.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import styled from "styled-components";
+
+import Box from "@src/components/UI/Box";
+import Icon from "@src/components/UI/Icon";
+
+const zIndex = 999999999;
```
[the remark this answers is not in the document]
Comment (ChiragJain-Haptik) [reply src/components/UI/EmptyScreen/StyledEmptyScreen.js]: Yes

**Review by sidMulay — DISMISSED** [review]

**Review by Neha0290 — APPROVED** [review]

File: src/components/UI/EmptyScreen/EmptyScreen.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (gopal-virtual) [reply src/components/UI/EmptyScreen/EmptyScreen.js:27]: print a friendly error if not passed.

File: src/components/UI/EmptyScreen/StyledEmptyScreen.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import styled from "styled-components";
```
Comment (gopal-virtual) [inline src/components/UI/EmptyScreen/StyledEmptyScreen.js:1]: Don't treat styled components as alien, they are part of the component, keep it in related files. Or name the file as helperComponent.

**Review by gopal-virtual — CHANGES_REQUESTED** [review]
This is very specific to Haptik, adding it to helpers would make more sense.

File: src/components/UI/EmptyScreen/StyledEmptyScreen.js (hunk 7 above)
Comment (ChiragJain-Haptik) [reply src/components/UI/EmptyScreen/StyledEmptyScreen.js]: Renamed the file

**Pull request closed without merging**